sdk/core/src/main/java/com/klaviyo/core/networking/KlaviyoNetworkMonitor.kt

```diff
@@ -53,6 +54,20 @@ internal object KlaviyoNetworkMonitor : NetworkMonitor {
         .getNetworkCapabilities(connectivityManager.activeNetwork)
         ?.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) ?: false
 
+    /**
+     * Check what type of network connection is currently servicing the device
+     *
+     * @return Integer representing the current network type
+     */
+    override fun getNetworkType(): Int {
```

Comment:
You could broadcast this value as part of the `onNetworkChange` event, i.e. update `NetworkObserver` typealias to take a second argument, network type. Then network monitor obersvers can get live updates.

Reply:
Then again I guess that would only be necessary if we were "interrupting" a background job to change timers when network changes, which we said we didn't want to do.

Reply:
Oh good point, I'll look into that!

Reply:
Yeah, or it would just be unused. So agreed, no need to make that change now.